### PR TITLE
[WIP][NOTEST] Sprout - Some bugfixes

### PR DESCRIPTION
**Pull request by mfalesni**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 5681a21b9f3dd3cdf39344640dd84624d01571af:

Everything seems all right from lint perspective :smile: .

Commit inspection report:
**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 078aea7d5ee1000c85f4f0661988b80f2091e5c0:

Everything seems all right from lint perspective :smile: .

Commit inspection report:
**No commit flaws detected**.

*CFME QE Bot*

**Comment by mfalesni** [comment]
@psav Please review, I will return back to this on monday because without this sprout has some issues. There are lots of places where it asked for current_appliance so I started going through it.

**Comment by dajoRH** [comment]
Lint report for commit d32976780c11dd7a70604c6e94b7ac615f75ed8c:

Everything seems all right from lint perspective :smile: .

Commit inspection report:
**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 30c42171e0d68b0ee3d1a5abc5b97d8f4cc2fd52:

Everything seems all right from lint perspective :smile: .

Commit inspection report:
**No commit flaws detected**.

*CFME QE Bot*

**Comment by mfalesni** [comment]
@psav This shall be it, asking for review. I tested it quickly, seems to work.

File: sprout/appliances/views.py [hunk 1]
```diff
@@ -84,6 +84,12 @@ def providers(request, provider_id=None):
     else:
         try:
             provider = Provider.objects.filter(id=provider_id, **user_filter).distinct().first()
+            if provider is None:
```
Comment (psav) [inline sprout/appliances/views.py]: ```if not provider```?

File: utils/appliance.py [hunk 2]
```diff
@@ -536,12 +536,12 @@ def patch_with_miqqe(self, log_callback=None):
 
         if patched_anything:
             logger.info("Cleaning and precompiling assets")
-            store.current_appliance.precompile_assets()
+            self.precompile_assets()
```
Comment (psav) [inline utils/appliance.py]: Yeh these are real important

**Comment by psav** [comment]
I'm thinking that actually we can get rid of the whole signal system, if you think about it, it was put in place before we had a lot of things objectized. As such, providers call a signal called "providers_changed". This actually doesn't seem to be hooked into anything, BUT, if it was, the provider object, would have an appliance linked to it as part of the new navigation efforts, so it could just do,
```self.appliance.providers_changed()``` and call a method directly on the appliance instead of firing off a signal.

The ```server_details_changed``` signal handled in the configure/configuration section, BasicInformation would also have an appliance object and hence wouldn't need signals. Appliance and DB both also fire this hook, appliance is self explanatory and I think db is only used inside the set_yaml_config which actually really doesn't need to be in here anymore and could just be in appliance.py.

```server_config_changed``` is again only used in two places, both of which would be ripe for removal and relate to both db and appliance.py modifications.

File: sprout/appliances/views.py (hunk 1 above)
Comment (mfalesni) [inline sprout/appliances/views.py]: .first() returns None if no object was returned.

**Comment by mfalesni** [comment]
Some testing comments:

**Comment by mfalesni** [comment]
@dajoRH please merge

**Comment by dajoRH** [comment]
Response to comment #244897700:

It seems like all the basic requirements were met, I have therefore merged your PR.

**Comment by dajoRH** [comment]
Response to comment #244897700:

The file cfme/configure/configuration/__init__.py is out of your allowed scope.
The file sprout/appliances/tasks.py is out of your allowed scope.
The file sprout/appliances/views.py is out of your allowed scope.
The file sprout/sprout/settings.py is out of your allowed scope.
The file utils/appliance.py is out of your allowed scope.
The file utils/db.py is out of your allowed scope.
The file utils/signals.py is out of your allowed scope.
Based on the PR analysis I cannot allow you to merge this PR on your own. Contact the CFME reviewers.

**Comment by jkrocil** [comment]
@dajoRH please merge

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/configure/configuration/__init__.py [hunk 3]
```diff
@@ -800,8 +792,7 @@ def update(self):
         sel.force_navigate("cfg_settings_currentserver_server")
         fill(self.basic_information, self.details)
         sel.click(form_buttons.save)
-        # TODO: Maybe make a cascaded delete on lazycache?
-        fire('server_details_changed')
+        store.current_appliance.server_details_changed()
```
Comment (psav) [inline cfme/configure/configuration/__init__.py]: I'd say go the extra step and get the self.appliance into the object, we'll need it for navigation anyway.

Comment (mfalesni) [inline cfme/configure/configuration/__init__.py]: Can it be a shim for now?

Comment (psav) [inline cfme/configure/configuration/__init__.py]: I guess, but it's two one liners to get appliance in there ;)

**Comment by dajoRH** [comment]
Lint report for commit fa6ab2997a024ff6356267d5704798487b697548:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

File: cfme/configure/configuration/__init__.py [hunk 4]
```diff
@@ -773,26 +773,26 @@ class BasicInformation(Updateable, Pretty):
             ('time_zone', Select("select#server_timezone")),
         ]
     )
-    pretty_attrs = ['company_name', 'appliance_name', 'appliance_zone', 'time_zone']
+    pretty_attrs = ['company_name', 'appliance_name', 'appliance_zone', 'time_zone', 'appliance']
```
Comment (psav) [inline cfme/configure/configuration/__init__.py]: Need this too
```python
appliance = CurrentAppliance()
```

File: cfme/configure/configuration/__init__.py [hunk 5]
```diff
@@ -773,26 +773,26 @@ class BasicInformation(Updateable, Pretty):
...
-            company_name=company_name,
-            appliance_name=appliance_name,
-            appliance_zone=appliance_zone,
-            time_zone=time_zone
-        )
+        self.company_name = company_name
+        self.appliance_name = appliance_name
+        self.appliance_zone = appliance_zone
+        self.time_zone = time_zone
+        self.appliance = appliance or store.current_appliance
```
Comment (psav) [inline cfme/configure/configuration/__init__.py]: Shuld be
```python
self.appliance = appliance or get_or_create_current_appliance()
```

**Comment by dajoRH** [comment]
Lint report for commit b487ecf34c8db23e4fd0dfb142641a71b092bb92:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 8e126818d11fd28b409da11302bf8103e47f3631:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 2a751f0509a6559bb8739c0f03ec088fbf1a786d:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 543fdc531f691f2eb31f85a6074ae2d638ed812b:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Pull request merged**